src/katgpucbf/xbgpu/bsend.py

```diff
@@ -267,17 +322,18 @@ def __init__(
             raise ValueError("channel_offset must be an integer multiple of n_channels_per_substream")
 
         self.tx_enabled = [tx_enabled] * len(outputs)
-        self.n_beams = len(outputs)
+        n_beams = len(outputs)
+        self.beam_names = [output.name for output in outputs]
```

Comment:
I made the beam names an attribute of the BSend as `Chunk.send` doesn't get called directly from the BPipeline
* It gets called from `BSend.send_chunk`